File: Assignment 0/Navya_191IT135/Anagram.java
import java.util.*;
class Anagram
{
    public static void main(String args[])
    {
        Scanner scan = new Scanner(System.in);
        System.out.println("Enter String1: ");
        String s1 = scan.nextLine();
        System.out.println("Enter String2: ");
        String s2 = scan.nextLine();
        int[] arr1 = new int[26];
        int[] arr2 = new int[26];
        check(arr1,s1);
        check(arr2,s2);
        int i;
        for(i = 0; i < 26; i++)
        {
            if(arr1[i]!=arr2[i])break;
        }
        if(i == 26)System.out.println("It is an anagram.");
        else System.out.println("It is not an anagram.");
    }
    static void check(int[] a,String s)
    {
       int i;
       for(i=0;i<s.length();i++)
       {
           int ch = s.charAt(i)-96;
           if(ch == 1)
           {
               a[1]++;
               continue;
           }
           int c1 = checkPrime(ch);
           int c2 = checkPrime(26-ch);
           if(c1 == 0&&c2 == 0)
           {
               a[ch]++;
               a[26-ch]++;
           }
       }
    }
    static int checkPrime(int x)
    {
        int flag = 0;
        for(int i = 2; i<=x/2; i++)
        {
            if(x%i==0)
            {
                flag = 1;
                break;
            }
        }
        return flag;
    }
}  

File: Assignment 0/Navya_191IT135/anagram.py
s1 = input("Enter string1: ")
print(s1)
arr = [ 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101]
s2 = input("Enter string2: ")
print(s2)
p1 = 1
for i in s1:
    x = ord(i)-97;
    p1*=arr[x]
print(str(p1))
p2 = 1
for i in s2:
    x = ord(i)-97;
    p2*=arr[x]
print(str(p2))
if p1==p2:
    print("It is an anagram.")
else:
    print("It is not an anagram")